Add date range filtering all verified endpoint

chriswininger opened the pull request:
Allow for optional filtering by date range

The verified endpoint now accepts an optional start and end date

you must provide both

when present we will only return events within that range

pagination will still apply

Digi-D committed:
Merge pull request #149 from infinite-industries/development

Production deploy 2020-05-10

Digi-D committed:
Merge pull request #150 from infinite-industries/development

Development

Digi-D committed:
Merge pull request #151 from infinite-industries/development

Quick fix to Card layouts issues

chriswininger committed:
Merge pull request #161 from infinite-industries/development

Release Develop

Digi-D committed:
Merge pull request #177 from infinite-industries/development

Update 2020-09-20

chriswininger committed:
Merge pull request #228 from infinite-industries/development

Release Develop

MatthewGidcomb committed:
Merge pull request #234 from infinite-industries/development

Release

chriswininger committed:
Merge pull request #248 from infinite-industries/development

Release development to production

chriswininger committed:
Merge pull request #253 from infinite-industries/development

Development

MatthewGidcomb committed:
Merge pull request #259 from infinite-industries/development

Release

MatthewGidcomb committed:
Merge pull request #264 from infinite-industries/development

Release

Digi-D committed:
Merge pull request #267 from infinite-industries/development

Update who we are with Becky Info and Dima image (#266)

chriswininger committed:
Merge pull request #273 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #281 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #287 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #291 from infinite-industries/development

Publish to docker hub (#290)

MatthewGidcomb committed:
Merge pull request #296 from infinite-industries/development

Release Development

MatthewGidcomb committed:
Merge pull request #301 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #308 from infinite-industries/development

Release development

chriswininger committed:
Release Development

Release Development

chriswininger committed:
Merge pull request #320 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #332 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #340 from infinite-industries/development

Release Development

chriswininger committed:
Merge pull request #363 from infinite-industries/development

Production release

chriswininger committed:
Merge pull request #389 from infinite-industries/development

Release Development

MatthewGidcomb committed:
Merge pull request #398 from infinite-industries/development

Production release

jswank committed:
Pull in recent changes to prod (#413)

- Updating NestJS and Sequelize
- add new workflow for container publication
- Add docker image used by database backup process (#404)
- Add MAP_BOX_API_KEY to api-server config (#399)
- Web Portal dependency updates (#405)
- Improve ansible var docs (#411)
- Add Codespaces support (#408)
- move docker images to ghcr.io (#410)
---------

Co-authored-by: Chris Wininger <chris.wininger@gmail.com>
Co-authored-by: Matthew Gidcomb <matthew.gidcomb@outlook.com>

MatthewGidcomb committed:
Revert "Pull in recent changes to prod (#413)" (#415)

This reverts commit 149a3d99235c2504fdfb69778a82d22b236552dd.

chriswininger committed:
Merge pull request #414 from infinite-industries/development

Production release

jswank committed:
Merge pull request #417 from infinite-industries/development

add prod fathom id (#416)

Digi-D committed:
Merge pull request #423 from infinite-industries/development

Pull in #418

Digi-D committed:
Merge pull request #427 from infinite-industries/development

Merge pull request #424

chriswininger committed:
Merge pull request #433 from infinite-industries/development

Release Development

MatthewGidcomb committed:
Merge pull request #440 from infinite-industries/development

Production Release

chriswininger committed:
Merge pull request #457 from infinite-industries/development

Production Release

MatthewGidcomb committed:
Merge pull request #463 from infinite-industries/development

Production Release

MatthewGidcomb committed:
Merge pull request #466 from infinite-industries/development

Production Release

chriswininger committed:
Merge pull request #468 from infinite-industries/development

release develop

chriswininger committed:
Merge pull request #474 from infinite-industries/development

Production Release

chriswininger committed:
Merge pull request #484 from infinite-industries/development

Release Develop

chriswininger committed:
Merge pull request #494 from infinite-industries/development

Release Development to Production

chriswininger committed:
Merge pull request #504 from infinite-industries/development

Production release

chriswininger committed:
Merge pull request #516 from infinite-industries/development

Release Prod

chriswininger committed:
Merge pull request #517 from infinite-industries/development

release development

chriswininger committed:
Merge pull request #519 from infinite-industries/development

Release Development

MatthewGidcomb committed:
Merge pull request #523 from infinite-industries/development

Production Release

Digi-D committed:
Merge pull request #528 from infinite-industries/development

Release Dev

chriswininger committed:
Allow for optional filtering by date range

The verified endpoint now accepts an optional start and end date

you must provide both

when present we will only return events within that range

pagination will still apply

chriswininger committed:
now compiling

chriswininger committed:
handle bad requests

chriswininger committed:
test passing

chriswininger committed:
apply prettier

chriswininger committed:
remove unused

chriswininger committed:
made the test more complete

MatthewGidcomb committed:
Merge pull request #531 from infinite-industries/development

Production Release

chriswininger committed:
Merge pull request #533 from infinite-industries/development

Production Release

MatthewGidcomb commented on api-server/src/events/events.controller.ts:
```diff
@@ -128,10 +128,34 @@ export class EventsController {
+    example: '2024-12-31T23:59:59.999Z',
+    required: false,
+    type: String,
+  })
```
If we always need to include both, what would you think of doing one `dateRange` param with both dates separated by `/` (the [interval](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals) format)?

```dateRange=2025-01-01T00:00:00Z/2025-06-30T23:59:59Z```

chriswininger replied:
:thinking: I haven't seen that used much so it makes me a bit nervous, but I was stumbling over that a bit too. It would be simpler if this were a post request, we'd just do json. We could do json in the query param too though i don't know how common that is.

if we went with the string separator approach we'd still have to have parse it and verify they sent both because the client could still leave one out though it does make the intent clearer

chriswininger replied:
the more I think on it the more I like it though I think I'll make the change

chriswininger committed:
tweak filter

now we we are inlcuding any events that have at least one date_time in
the range

chriswininger committed:
further simplification of filter

chriswininger committed:
drop a few chatty comments that the LLMs made

chriswininger committed:
Merge branch 'master' into add-date-range-filtering-all-verified-endpoint

chriswininger committed:
Merge branch 'development' into add-date-range-filtering-all-verified-endpoint
